x/ecocredit/base/keeper/features/msg_update_project_enrollment.feature

```diff
@@ -0,0 +1,95 @@
+Feature: Msg/UpdateProjectEnrollment
+
+  Background:
+    Given project "P001"
+    * class "C01"
+    * class issuer "I01" for "C01"
+    * class issuer "I02" for "C01"
+
+  Rule: valid state transitions performed by issuers which don't remove enrollment entries are:
+  UNSPECIFIED -> CHANGES_REQUESTED or ACCEPTED
+  CHANGES_REQUESTED -> ACCEPTED
+  ACCEPTED -> ACCEPTED with new metadata
+    Scenario Outline: <scenario>
+      Given enrollment for "P001" to "C01" is "<cur_status>" with metadata "<cur_metadata>"
+      When "<issuer>" updates enrollment for "P001" to "C01" with status "<new_status>" and metadata "<new_metadata>"
+      Then expect error contains "<err>"
+      And expect enrollment for "P001" to "C01" to be "<expected_status>" with metadata "<expected_metadata>"
+      And if no error expect EventUpdateProjectEnrollment with properties
+      """
+        {
+            "issuer": "<issuer>", "project_id": "P001", "class_id": "C01",
+            "old_status": "PROJECT_ENROLLMENT_STATUS_<cur_status>",
+            "new_status": "PROJECT_ENROLLMENT_STATUS_<expected_status>",
+            "new_enrollment_metadata": "<expected_metadata>"
+        }
+      """
+      Examples:
+        | scenario                          | cur_status        | cur_metadata | issuer | new_status | new_metadata | err          | expected_status   | expected_metadata |
+        | I01 unspecified to accepted       | UNSPECIFIED       | abc          | I01    | ACCEPTED   | foo123       |              | ACCEPTED          | foo123            |
+        | Bob unspecified to accepted       | UNSPECIFIED       | abc          | Bob    | ACCEPTED   | foo123       | unauthorized | UNSPECIFIED       | abc               |
+        | I01 changes requested to accepted | CHANGES_REQUESTED | bar456       | I01    | ACCEPTED   | foo123       |              | ACCEPTED          | foo123            |
+        | Bob changes requested to accepted | CHANGES_REQUESTED | bar456       | Bob    | ACCEPTED   | foo123       | unauthorized | CHANGES_REQUESTED | bar456            |
+        | I01 update accepted metadata      | ACCEPTED          | foo123       | I01    | ACCEPTED   | bar357       |              | ACCEPTED          | bar357            |
+        | Bob updated accepted metadata     | ACCEPTED          | foo123       | Bob    | ACCEPTED   | bar357       | unauthorized | ACCEPTED          | foo123            |
+
+  Rule: valid state transitions performed by issuers which remove enrollment entries are:
+  UNSPECIFIED -> REJECTED
+  CHANGES_REQUESTED -> REJECTED
+  ACCEPTED -> TERMINATED
```

Comment:
Just want to double check if there are any issues having a project that is `TERMINATED` be removed from the project enrollment state. Considering that a project may have have credit batches issued while it was `ACCEPTED`. I checked the implementation to make sure we can still query those credit batches and it seems like that query would still work (not changed in this PR). Are there any other cases to consider?

Reply:
I think it's okay because removing it will allow it to re-enroll.